src/main/java/com/iConomy/system/Holdings.java

```diff
@@ -116,17 +118,35 @@ public void set(double balance) {
     }
 
     public void add(double amount) {
-        double balance = get();
-        double ending = balance + amount;
-
-        math(amount, balance, ending);
+        if (Bukkit.isPrimaryThread()) {
+            double balance = get();
+            double ending = balance + amount;
+    
+            math(amount, balance, ending);
+        }else{
+            new BukkitRunnable(){
+                @Override
+                public void run(){
+                    add(amount);
+                }
+            }.runTask(iConomy.instance);
+        }
     }
 
     public void subtract(double amount) {
-        double balance = get();
-        double ending = balance - amount;
-
-        math(amount, balance, ending);
+        if (Bukkit.isPrimaryThread()) {
+            double balance = get();
+            double ending = balance - amount;
+    
+            math(amount, balance, ending);
+        }else{
+            new BukkitRunnable(){
+                @Override
+                public void run(){
+                    add(amount);
```

Comment:
Looks to be a typo/copy-paste mix-up but this should be subtract(amount), no?

Reply:
oh yes!